### DISCUSSION TITLE
Integrate SAML SSO with argo workflow.

### DISCUSSION THREAD

**Discussion by gouravjoshicldcvr** [argoproj/argo-workflows#7452] [Q&A] [answered]
Hi Team,

I have done the SSO integration using SAML with argocd using below doc.
  https://argo-cd-docs.readthedocs.io/en/latest/operator-manual/sso/okta/#saml-with-dex

Same thing i want to achieve with argo-workflow UI as well.
i tried to find out but in the end i found only doc related to this, but this requires a client ID and client secret which is supported by OIDC but i need to use SAML.
  https://argoproj.github.io/argo-workflows/argo-server-sso-argocd/

Can anyone help me in this.
Thanks

**Comment by ugurcancaykara** [comment]
Any updates on this ?

**Comment by agilgur5** [comment, marked as answer]
You'll need to use Dex (or similar) to configure the SAML <-> OIDC exchange.

The SSO + Dex docs you linked, https://argoproj.github.io/argo-workflows/argo-server-sso-argocd/, explain how to do that and I have configured SAML that way in the past.
It can also be any Dex instance, does not necessarily have to be Argo CD's.

The Argo CD Okta docs you linked is how you integrate Okta with Dex: https://argo-cd-docs.readthedocs.io/en/latest/operator-manual/sso/okta/#saml-with-dex.

You _then_ follow the Workflows SSO + Dex docs, https://argoproj.github.io/argo-workflows/argo-server-sso-argocd/, to add the Workflows static client to Dex.

In this set-up, Workflows authenticates using Dex via OIDC, and Dex connects to Okta via SAML.
